### PR TITLE
Add an .editorconfig file

### PR DESCRIPTION
**Pull request by KindDragon**
EditorConfig allow all developers use same space/tab settings for all files if they install EditorConfig plugin

Similar to that PR https://github.com/dotnet/coreclr/pull/172

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+http://EditorConfig.org
+
+# top-most EditorConfig file
+root = true
+
+# Default settings:
+# A newline ending every file
+# Use 4 spaces as indentation
+[*]
+insert_final_newline = true
```
Comment (sharwell) [inline .editorconfig]: I recommend omitting this line, unless you want to see pull requests containing end-of-file newline changes that are completely unrelated to the topic of the pull request. For example, any time someone manually edits a project file it will add the newline. Then anytime someone edits the project file automatically (such as adding a new file within Visual Studio), it will remove it again.

**Comment by tmeschter** [comment]
:+1:

**Comment by tmeschter** [comment]
We also have some .vsct files--these are XML files that define Visual Studio commands and specify where they show up in the menus. These should use an indentation of two spaces, as well.

**Comment by jaredpar** [comment]
Can we change this to have the same contents as the corefx version?  We have the same coding guidelines as them so seemingly it should match.  

> https://github.com/dotnet/corefx/blob/master/.editorconfig

@terrajobst maybe we should add this to codeformatter as well?  It would make sense as the central place for style utils.

**Comment by KindDragon** [comment]
They use
insert_final_newline = true
and
trim_trailing_whitespace = true.

Maybe update their .editconfig in accordance with our .editconfig

**Comment by sharwell** [comment]
@jaredpar It should be the other way around. I missed the pull request to add that to corefx so I wasn't able to mention two corrections it needs.

**Comment by jaredpar** [comment]
@KindDragon 

The `insert_final_newline` value definitely needs to be changed.  It goes against what the [codeformatter](https://github.com/dotnet/codeformatter) will do.

**Comment by sharwell** [comment]
@jaredpar It's hard enough for most people to make sure that they only submit pull requests that change the lines of code relevant to the specific subject of the pull request. Options like `trim_trailing_whitespace` can automatically change unrelated lines of the file when the file is saved. If all of the source files in the repository *already* don't have trailing whitespace then it's not a problem. Otherwise it can really be a nuisance for people working to keep pull requests clean.

Plus, Visual Studio already does a pretty good job of taming trailing whitespace.

**Comment by jaredpar** [comment]
@sharwell the expectation though is that the code is always clean.  If it's not then it's a bug that I'm happy to see get pointed out when others submit PRs

**Comment by KindDragon** [comment]
> The trim_trailing_whitespace value appears to be correct though. Why do you feel it is incorrect?

It's already cause some troubles https://github.com/dotnet/coreclr/pull/172#issuecomment-74017261

**Comment by KindDragon** [comment]
> The insert_final_newline value definitely needs to be changed. It goes against what the codeformatter will do.

Codeformatter removes final empty line?

**Comment by KindDragon** [comment]
I prefer 4 spaces by default, but can change this to 2 spaces if you want.

**Comment by sharwell** [comment]
The `insert_final_newline` setting should simply be omitted, which means "keep things as they are". Otherwise there will be a never-ending battle between this setting and certain Visual Studio auto-save features.

**Comment by KindDragon** [comment]
That setting already removed from file

**Comment by sharwell** [comment]
@KindDragon I think I posted that comment on the wrong pull request. It's deleted now.

**Comment by KindDragon** [comment]
So, what's missing to accept this PR?

**Comment by sharwell** [comment]
@KindDragon it looks fine to me, but I'm not the one you have to convince :smile_cat:  (I don't actually merge things)

**Comment by terrajobst** [comment]
> @terrajobst maybe we should add this to codeformatter as well? It would make sense as the central place for style utils.

Good idea.
